lib/docker-delta.coffee

```diff
@@ -28,30 +28,32 @@ exports.createDelta = (srcImage, destImage, v2 = true) ->
 	# promises are in progress
 	deltaStream = new stream.PassThrough()
 
-	# We first get the two root directories and then apply rsync on them
-	rsyncStream = Promise.resolve [ srcImage, destImage ]
-		.bind(docker)
-		.map(docker.imageRootDir)
-		.map (rootDir) ->
-			path.join(rootDir, '/')
-		.spread(rsync.createRsyncStream)
-
-	# We also retrieve the container config for the image
+	# We retrieve the container config for the image
 	config = docker.getImage(destImage).inspectAsync().get('Config')
 
-	Promise.all [ config, rsyncStream ]
-	.spread (config, rsyncStream) ->
-		if v2
-			metadata =
-				version: 2
-				dockerConfig: config
-
-			# Write the header of the delta format which is the serialised metadata
-			deltaStream.write(JSON.stringify(metadata))
-			# Write the NUL byte separator for the rsync binary stream
-			deltaStream.write(new Buffer([ 0x00 ]))
-		# Write the rsync binary stream
-		rsyncStream.pipe(deltaStream)
+	# We then get the two root directories and then apply rsync on them
+	rootDirFunc = docker.imageRootDirMounted.bind(docker)
+	Promise.using [ srcImage, destImage ].map(rootDirFunc), (rootDirs) ->
+		[ srcDir, destDir ] = rootDirs.map (rootDir) -> path.join(rootDir, '/')
+		rsyncStream = rsync.createRsyncStream(srcDir, destDir)
+
+		Promise.all [ config, rsyncStream ]
+		.spread (config, rsyncStream) ->
+			if v2
+				metadata =
+					version: 2
+					dockerConfig: config
+
+				# Write the header of the delta format which is the serialised metadata
+				deltaStream.write(JSON.stringify(metadata))
+				# Write the NUL byte separator for the rsync binary stream
+				deltaStream.write(new Buffer([ 0x00 ]))
+			# Write the rsync binary stream
+			new Promise (resolve, reject) ->
+				rsyncStream
+				.on 'error', reject
+				.on 'close', resolve
```

Comment:
These error handlers are the only change in this big diff block. Can we only have those in the PR and keep the rest as-is?